iml-manager-cli/src/server.rs

```diff
@@ -349,7 +375,11 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
             let objects: Vec<_> = objects
                 .into_iter()
                 .filter_map(|x| x.host_profiles)
-                .filter(|x| is_profile_valid(&x, &profile.name))
+                .filter(|x| is_profile_exist(&x, &profile.name))
```

Comment:
This should be a hard error. If the profile doesn't exist, we should print an error and return. Something like:

"Profile {} not found".

This means using `find` at this point and handling the `Some` / `None`  variants.

Reply:
So if one of the objects doesn't have a profile, we stop the whole processing, right?

Reply:
Right If we can't find the profile we are looking for in this iterator, we return early with an error.

Reply:
So each element in objects must have a profile to start the next step?

Reply:
There should only be one match, and yes this is different from current logic where we filtered non-matches previously

Reply:
Ok. New Year is a holiday for my family so I'll make a patch by the 2nd.

Reply:
I have put an updated version. 
Please notice that the logic has been changed so it needs to be checked again.